src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstancesLoaderImpl.scala

```diff
@@ -25,12 +25,7 @@ private[tracker] class InstancesLoaderImpl(repo: InstanceRepository)(implicit va
       instances <- Future.sequence(names.map(repo.get)).map(_.flatten)
     } yield {
       log.info(s"Loaded ${instances.size} tasks")
-      val instancesByRunSpec = instances.groupBy(_.runSpecId)
-      val map = instancesByRunSpec.map {
-        case (runSpecId, specInstances) => runSpecId ->
-          InstanceTracker.SpecInstances.forInstances(runSpecId, specInstances)
```

Comment:
why did we have to it like this previously?

Reply:
This was just code duplication that got in the base with the infamous `Team` merge for pods.